dev/tests/static/testsuite/Magento/Test/Integrity/Dependency/GraphQlSchemaDependencyProvider.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Test\Integrity\Dependency;
+
+use Magento\Framework\App\Bootstrap;
+use Magento\Framework\GraphQlSchemaStitching\GraphQlReader;
+use Magento\Framework\GraphQlSchemaStitching\GraphQlReader\TypeReaderComposite;
+
+/**
+ * Provide information on the dependency between the modules according to the GraphQL schema.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ */
+class GraphQlSchemaDependencyProvider extends DependencyProvider
+{
+    /**
+     * @var array
+     */
+    private $parsedSchema = [];
+
+    /**
+     * GraphQlSchemaDependencyProvider constructor.
+     */
+    public function __construct()
+    {
+        $this->getGraphQlSchemaDeclaration();
+    }
+
+    /**
+     * Provide declared dependencies between modules based on the declarative schema configuration.
+     *
+     * @param string $moduleName
+     * @return array
+     * @throws \Exception
+     */
+    public function getDeclaredExistingModuleDependencies(string $moduleName): array
+    {
+        $this->initDeclaredDependencies();
+        $dependencies = $this->getDependenciesFromSchema($moduleName);
+        $declared = $this->getDeclaredDependencies($moduleName, self::TYPE_HARD, self::MAP_TYPE_DECLARED);
+        return array_unique(array_values(array_intersect($declared, $dependencies)));
+    }
+
+    /**
+     * Provide undeclared dependencies between modules based on the declarative schema configuration.
+     *
+     * [
+     *     $dependencyId => [$module1, $module2, $module3 ...],
+     *     ...
+     * ]
+     *
+     * @param string $moduleName
+     * @return array
+     * @throws \Exception
```

Comment:
should not throw simple exceptions
https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html#5-exceptions